swarm -g 480 -t 24 -f last.swarm --partition largemem --time 10:00:00 --module plink,R

### Polygenic risk score analyses of 119 bacteria genuses versus PD risk

## Make a list of summary stats file 
ls *csv > genuses.txt ## These files have been previously pruned by p-value < 0.05

## Format these files
cat genuses.txt | while read LINE 
do
  echo $LINE
  sed 's/\"//g' $LINE | sed 's/,/ /g' > temp.txt
  awk '{print $0"\t"$2":"$3}' temp.txt | sed 's/chr\:bp/ID/' > $LINE.temp_formatted.txt
  rm temp.txt
done

## Re-format files 
cat genuses.txt | while read LINE 
do
  echo $LINE
  awk '{print $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14}' $LINE.temp_formatted.txt > $LINE.temp2_formatted.txt
done

## Re-format files 
cat genuses.txt | while read LINE 
do
  echo $LINE
  grep "rs" $LINE.temp2_formatted.txt > $LINE.temp_formatted_final.txt
done

## Identify independent risk SNPs using our in-house LD reference data for European populations (/data/LNG/pdMeta5v2/cojoRef/binaryForCojo) - default clumping threshold (r2 = 0.1, kb= 250kb) + 10000 permutations
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Actinomyces.id.423.summary.txt.csv.temp_formatted_final.txt --out Actinomyces.id.423
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Adlercreutzia.id.812.summary.txt.csv.temp_formatted_final.txt --out Adlercreutzia.id.812
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Akkermansia.id.4037.summary.txt.csv.temp_formatted_final.txt --out Akkermansia.id.4037
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Alistipes.id.968.summary.txt.csv.temp_formatted_final.txt --out Alistipes.id.968
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Allisonella.id.2174.summary.txt.csv.temp_formatted_final.txt --out Allisonella.id.2174
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Alloprevotella.id.961.summary.txt.csv.temp_formatted_final.txt --out Alloprevotella.id.961
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Anaerofilum.id.2053.summary.txt.csv.temp_formatted_final.txt --out Anaerofilum.id.2053
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Anaerostipes.id.1991.summary.txt.csv.temp_formatted_final.txt --out Anaerostipes.id.1991
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Anaerotruncus.id.2054.summary.txt.csv.temp_formatted_final.txt --out Anaerotruncus.id.2054
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Bacteroides.id.918.summary.txt.csv.temp_formatted_final.txt --out Bacteroides.id.918
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Barnesiella.id.944.summary.txt.csv.temp_formatted_final.txt --out Barnesiella.id.944
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Bifidobacterium.id.436.summary.txt.csv.temp_formatted_final.txt --out Bifidobacterium.id.436
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Bilophila.id.3170.summary.txt.csv.temp_formatted_final.txt --out Bilophila.id.3170
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Blautia.id.1992.summary.txt.csv.temp_formatted_final.txt --out Blautia.id.1992
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Butyricicoccus.id.2055.summary.txt.csv.temp_formatted_final.txt --out Butyricicoccus.id.2055
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Butyricimonas.id.945.summary.txt.csv.temp_formatted_final.txt --out Butyricimonas.id.945
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Butyrivibrio.id.1993.summary.txt.csv.temp_formatted_final.txt --out Butyrivibrio.id.1993
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.CandidatusSoleaferrea.id.11350.summary.txt.csv.temp_formatted_final.txt --out CandidatusSoleaferrea.id.11350
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Catenibacterium.id.2153.summary.txt.csv.temp_formatted_final.txt--out Catenibacterium.id.2153
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.ChristensenellaceaeR.7group.id.11283.summary.txt.csv.temp_formatted_final.txt --out ChristensenellaceaeR.7group.id.11283
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Clostridiuminnocuumgroup.id.14397.summary.txt.csv.temp_formatted_final.txt --out Clostridiuminnocuumgroup.id.14397
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Clostridiumsensustricto1.id.1873.summary.txt.csv.temp_formatted_final.txt --out Clostridiumsensustricto1.id.1873
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Collinsella.id.815.summary.txt.csv.temp_formatted_final.txt --out Collinsella.id.815
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Coprobacter.id.949.summary.txt.csv.temp_formatted_final.txt --out Coprobacter.id.949
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Coprococcus1.id.11301.summary.txt.csv.temp_formatted_final.txt --out Coprococcus1.id.11301
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Coprococcus2.id.11302.summary.txt.csv.temp_formatted_final.txt --out Coprococcus2.id.11302
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Coprococcus3.id.11303.summary.txt.csv.temp_formatted_final.txt --out Coprococcus3.id.11303
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.DefluviitaleaceaeUCG011.id.11287.summary.txt.csv.temp_formatted_final.txt --out DefluviitaleaceaeUCG011.id.11287
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Desulfovibrio.id.3173.summary.txt.csv.temp_formatted_final.txt --out Desulfovibrio.id.3173
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Dialister.id.2183.summary.txt.csv.temp_formatted_final.txt --out Dialister.id.2183
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Dorea.id.1997.summary.txt.csv.temp_formatted_final.txt --out Dorea.id.1997
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Eggerthella.id.819.summary.txt.csv.temp_formatted_final.txt --out Eggerthella.id.819
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Eisenbergiella.id.11304.summary.txt.csv.temp_formatted_final.txt --out Eisenbergiella.id.11304
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Enterorhabdus.id.820.summary.txt.csv.temp_formatted_final.txt --out Enterorhabdus.id.820
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Erysipelatoclostridium.id.11381.summary.txt.csv.temp_formatted_final.txt--out Erysipelatoclostridium.id.11381
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.ErysipelotrichaceaeUCG003.id.11384.summary.txt.csv.temp_formatted_final.txt --out ErysipelotrichaceaeUCG003.id.11384
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Escherichia.Shigella.id.3504.summary.txt.csv.temp_formatted_final.txt --out Escherichia.Shigella.id.3504
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumbrachygroup.id.11296.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumbrachygroup.id.11296
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumcoprostanoligenesgroup.id.11375.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumcoprostanoligenesgroup.id.11375
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumeligensgroup.id.14372.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumeligensgroup.id.14372
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumfissicatenagroup.id.14373.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumfissicatenagroup.id.14373
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumhalliigroup.id.11338.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumhalliigroup.id.11338
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumnodatumgroup.id.11297.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumnodatumgroup.id.11297
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumoxidoreducensgroup.id.11339.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumoxidoreducensgroup.id.11339
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumrectalegroup.id.14374.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumrectalegroup.id.14374
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumruminantiumgroup.id.11340.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumruminantiumgroup.id.11340
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumventriosumgroup.id.11341.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumventriosumgroup.id.11341
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Eubacteriumxylanophilumgroup.id.14375.summary.txt.csv.temp_formatted_final.txt --out .Eubacteriumxylanophilumgroup.id.14375
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Faecalibacterium.id.2057.summary.txt.csv.temp_formatted_final.txt --out Faecalibacterium.id.2057
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.FamilyXIIIAD3011group.id.11293.summary.txt.csv.temp_formatted_final.txt --out FamilyXIIIAD3011group.id.11293
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.FamilyXIIIUCG001.id.11294.summary.txt.csv.temp_formatted_final.txt --out FamilyXIIIUCG001.id.11294
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Flavonifractor.id.2059.summary.txt.csv.temp_formatted_final.txt --out Flavonifractor.id.2059
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Fusicatenibacter.id.11305.summary.txt.csv.temp_formatted_final.txt --out Fusicatenibacter.id.11305
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Gordonibacter.id.821.summary.txt.csv.temp_formatted_final.txtt --out Gordonibacter.id.821
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Haemophilus.id.3698.summary.txt.csv.temp_formatted_final.txt --out Haemophilus.id.3698
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Holdemanella.id.11393.summary.txt.csv.temp_formatted_final.txt --out Holdemanella.id.11393
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Holdemania.id.2157.summary.txt.csv.temp_formatted_final.txt --out Holdemania.id.2157
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Howardella.id.2000.summary.txt.csv.temp_formatted_final.txt --out Howardella.id.2000
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Hungatella.id.11306.summary.txt.csv.temp_formatted_final.txt --out Hungatella.id.11306
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Intestinibacter.id.11345.summary.txt.csv.temp_formatted_final.txt--out Intestinibacter.id.11345
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Intestinimonas.id.2062.summary.txt.csv.temp_formatted_final.txt --out Intestinimonas.id.2062
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Lachnoclostridium.id.11308.summary.txt.csv.temp_formatted_final.txt --out Lachnoclostridium.id.11308
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeFCS020group.id.11314.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeFCS020group.id.11314
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeNC2004group.id.11316.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeNC2004group.id.11316
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeND3007group.id.11317.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeND3007group.id.11317
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeNK4A136group.id.11319.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeNK4A136group.id.11319
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeUCG001.id.11321.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeUCG001.id.11321
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeUCG004.id.11324.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeUCG004.id.11324
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeUCG008.id.11328.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeUCG008.id.11328
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.LachnospiraceaeUCG010.id.11330.summary.txt.csv.temp_formatted_final.txt --out LachnospiraceaeUCG010.id.11330
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Lachnospira.id.2004.summary.txt.csv.temp_formatted_final.txt --out Lachnospira.id.2004
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Lactobacillus.id.1837.summary.txt.csv.temp_formatted_final.txt --out Lactobacillus.id.1837
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Lactococcus.id.1851.summary.txt.csv.temp_formatted_final.txt --out Lactococcus.id.1851
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Marvinbryantia.id.2005.summary.txt.csv.temp_formatted_final.txt --out Marvinbryantia.id.2005
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Methanobrevibacter.id.123.summary.txt.csv.temp_formatted_final.txt--out Methanobrevibacter.id.123
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Odoribacter.id.952.summary.txt.csv.temp_formatted_final.txt --out Odoribacter.id.952
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Olsenella.id.822.summary.txt.csv.temp_formatted_final.txt --out Olsenella.id.822
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Oscillibacter.id.2063.summary.txt.csv.temp_formatted_final.txt --out Oscillibacter.id.2063
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Oscillospira.id.2064.summary.txt.csv.temp_formatted_final.txt --out Oscillospira.id.2064
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Oxalobacter.id.2978.summary.txt.csv.temp_formatted_final.txt--out Oxalobacter.id.2978
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Parabacteroides.id.954.summary.txt.csv.temp_formatted_final.txt --out Parabacteroides.id.954
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Paraprevotella.id.962.summary.txt.csv.temp_formatted_final.txt --out Paraprevotella.id.962
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Parasutterella.id.2892.summary.txt.csv.temp_formatted_final.txt --out Parasutterella.id.2892
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Peptococcus.id.2037.summary.txt.csv.temp_formatted_final.txt --out Peptococcus.id.2037
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Phascolarctobacterium.id.2168.summary.txt.csv.temp_formatted_final.txt --out Phascolarctobacterium.id.2168
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Prevotella7.id.11182.summary.txt.csv.temp_formatted_final.txt --out Prevotella7.id.11182
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Prevotella9.id.11183.summary.txt.csv.temp_formatted_final.txt --out Prevotella9.id.11183
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RikenellaceaeRC9gutgroup.id.11191.summary.txt.csv.temp_formatted_final.txt --out RikenellaceaeRC9gutgroup.id.11191
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Romboutsia.id.11347.summary.txt.csv.temp_formatted_final.txt --out Romboutsia.id.11347
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Roseburia.id.2012.summary.txt.csv.temp_formatted_final.txt --out Roseburia.id.2012
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Ruminiclostridium5.id.11355.summary.txt.csv.temp_formatted_final.txt --out Ruminiclostridium5.id.11355
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Ruminiclostridium6.id.11356.summary.txt.csv.temp_formatted_final.txt --out Ruminiclostridium6.id.11356
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Ruminiclostridium9.id.11357.summary.txt.csv.temp_formatted_final.txt --out Ruminiclostridium9.id.11357
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeNK4A214group.id.11358.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeNK4A214group.id.11358
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG002.id.11360.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG002.id.11360
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG003.id.11361.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG003.id.11361
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG004.id.11362.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG004.id.11362
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG005.id.11363.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG005.id.11363
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG009.id.11366.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG009.id.11366
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG010.id.11367.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG010.id.11367
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG011.id.11368.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG011.id.11368
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG013.id.11370.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG013.id.11370
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.RuminococcaceaeUCG014.id.11371.summary.txt.csv.temp_formatted_final.txt --out RuminococcaceaeUCG014.id.11371
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Ruminococcus1.id.11373.summary.txt.csv.temp_formatted_final.txt --out Ruminococcus1.id.11373
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Ruminococcus2.id.11374.summary.txt.csv.temp_formatted_final.txt --out Ruminococcus2.id.11374
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Ruminococcusgauvreauiigroup.id.11342.summary.txt.csv.temp_formatted_final.txt --out .Ruminococcusgauvreauiigroup.id.11342
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Ruminococcusgnavusgroup.id.14376.summary.txt.csv.temp_formatted_final.txt --out .Ruminococcusgnavusgroup.id.14376
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus..Ruminococcustorquesgroup.id.14377.summary.txt.csv.temp_formatted_final.txt --out .Ruminococcustorquesgroup.id.14377
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Sellimonas.id.14369.summary.txt.csv.temp_formatted_final.txt --out Sellimonas.id.14369
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Senegalimassilia.id.11160.summary.txt.csv.temp_formatted_final.txt --out Senegalimassilia.id.11160
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Slackia.id.825.summary.txt.csv.temp_formatted_final.txt --out Slackia.id.825
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Streptococcus.id.1853.summary.txt.csv.temp_formatted_final.txt --out Streptococcus.id.1853
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Subdoligranulum.id.2070.summary.txt.csv.temp_formatted_final.txt --out Subdoligranulum.id.2070
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Sutterella.id.2896.summary.txt.csv.temp_formatted_final.txt --out Sutterella.id.2896
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Terrisporobacter.id.11348.summary.txt.csv.temp_formatted_final.txt --out Terrisporobacter.id.11348
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Turicibacter.id.2162.summary.txt.csv.temp_formatted_final.txt --out Turicibacter.id.2162
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Tyzzerella3.id.11335.summary.txt.csv.temp_formatted_final.txt --out Tyzzerella3.id.11335
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Veillonella.id.2198.summary.txt.csv.temp_formatted_final.txt --out Veillonella.id.2198
Rscript /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice.R --cov-file /data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab -t /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --beta --snp ID --A1 eff.allele --A2 ref.allele --stat beta --se SE --pvalue pDerived --ld /data/LNG/pdMeta5v2/cojoRef/binaryForCojo  --print-snp --score std --perm 10000 --prsice /data/LNG/pdMeta5v2/leaveOneOutPrsice/PRSice_linux/PRSice_linux -n 8 --binary-target T --quantile 4 --prevalence 0.005 --fastscore -b addedPgenus.Victivallis.id.2256.summary.txt.csv.temp_formatted_final.txt --out Victivallis.id.2256

## Remove NeuroX individuals & extract nominated variants
cat genuses_formatted_list.txt | while read LINE 
do
plink --bfile /data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins --remove-fam NeuroX.txt --extract $LINE.snps --make-bed --out pruned_$LINE
done

## Make score files
cat genuses_formatted_list.txt | while read LINE 
do
awk '{print $14, $6, $7}' addedPgenus.$LINE.summary.txt.csv.temp_formatted.txt | sed '1d' > $LINE.toscore.txt
done

## Make sure score files have 3 expected fields rather than 2
cat genuses_formatted_list.txt | while read LINE 
do
grep ":" $LINE.toscore.txt > true_$LINE.toscore.txt
done

##awk: fatal: cannot open file `addedPgenus.Clostridiuminnocuumgroup.id.14397.summary.txt.csv.temp_formatted.txt' for reading (No such file or directory)

## Calculate scores
cat genuses_formatted_list.txt | while read LINE 
do
plink --bfile pruned_$LINE --score true_$LINE.toscore.txt --make-bed --out pruned_$LINE
done

ls *.profile* > list_profiles.txt

## Run PRS (logistic regression) on R - versus PD risk
R
library("data.table")
listOfProfiles <- read.table("list_profiles.txt", header = T)
names(listOfProfiles) <- c("id")
covs1 <- fread("/data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab", header = T)
covs2 <- fread("/data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins.fam", header = F)
colnames(covs2) <- c("FID", "IID", "MAT", "PAT", "SEX", "PHENO")
covsfinal <- merge (covs1, covs2, by ="FID")
covsfinal$CASE <- covsfinal$PHENO - 1
outPut <- matrix(ncol = 4, nrow = length(listOfProfiles$id), NA)
colnames(outPut) <- c("genus","b","se","p")
for(i in 1:length(listOfProfiles$id))
{
	profileName <- 	as.character(listOfProfiles$id[i])
	profile <- fread(file = paste(profileName, ".profile", sep = ""), header = T)
	profile$index <- paste(profile$FID, profile$IID, sep = "")
	data <- merge(covsfinal, profile, by = "FID")
	meanControls <- mean(data$SCORE[data$CASE == 0])
	sdControls <- sd(data$SCORE[data$CASE == 0])
	data$zSCORE <- (data$SCORE - meanControls)/sdControls	
	grsTest <- glm(CASE ~ zSCORE + SEX + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, family="binomial", data = data)	
	beta <- summary(grsTest)$coefficients["zSCORE","Estimate"]
	se <- summary(grsTest)$coefficients["zSCORE","Std. Error"]
	p <- summary(grsTest)$coefficients["zSCORE","Pr(>|z|)"]
	outPut[i,1] <- profileName
	outPut[i,2] <- beta
	outPut[i,3] <- se
	outPut[i,4] <- p
}
write.table(outPut, "Genus_PRS.tab", quote = F, sep = "\t", row.names = F)

## Run PRS (linear regression) on R - versus AGE AT ONSET
R
library("data.table")
listOfProfiles <- read.table("list_profiles.txt", header = T)
names(listOfProfiles) <- c("id")
covs1 <- fread("/data/LNG/saraB/WGS/toPRSice_phenosAndCovs_renamed.tab", header = T)
covs2 <- fread("/data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins.fam", header = F)
colnames(covs2) <- c("FID", "IID", "MAT", "PAT", "SEX", "PHENO")
covsfinal <- merge (covs1, covs2, by ="FID")
covsfinal$CASE <- covsfinal$PHENO - 1
outPut <- matrix(ncol = 4, nrow = length(listOfProfiles$id), NA)
colnames(outPut) <- c("genus","b","se","p")
for(i in 1:length(listOfProfiles$id))
{
	profileName <- 	as.character(listOfProfiles$id[i])
	profile <- fread(file = paste(profileName, ".profile", sep = ""), header = T)
	profile$index <- paste(profile$FID, profile$IID, sep = "")
	data <- merge(covsfinal, profile, by = "FID")
	cases <- subset(data, CASE == 1)
	meanPop <- mean(cases$SCORE)
	sdPop <- sd(cases$SCORE)
	cases$zSCORE <- (cases$SCORE - meanPop)/sdPop
	grsTest <- lm(AGE ~ zSCORE + SEX + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, data = cases)	
	beta <- summary(grsTest)$coefficients["zSCORE","Estimate"]
	se <- summary(grsTest)$coefficients["zSCORE","Std. Error"]
	p <- summary(grsTest)$coefficients["zSCORE","Pr(>|t|)"]
	outPut[i,1] <- profileName
	outPut[i,2] <- beta
	outPut[i,3] <- se
	outPut[i,4] <- p
}
write.table(outPut, "Genus_PRS_AGE.tab", quote = F, sep = "\t", row.names = F)


## TEST ANALYSIS for Actinomyces

# library("data.table")
# temp_data <- read.table("pruned_Actinomyces.id.423.profile", header = T) 
# covs1 <- fread("/data/LNG/saraB/WGS/noage_toPRSice_phenosAndCovs_renamed.tab", header = T)
# covs2 <- fread("/data/LNG/saraB/concat_HARDCALLS_PD_september_2018_no_cousins.fam", header = F)
# colnames(covs2) <- c("FID", "IID", "MAT", "PAT", "SEX", "PHENO")
# covsfinal <- merge (covs1, covs2, by ="FID")
# data <- merge(temp_data, covsfinal, by = "FID")
# data$CASE <- data$PHENO.x - 1
# meanControls <- mean(data$SCORE[data$CASE == 0])
# sdControls <- sd(data$SCORE[data$CASE == 0])
# data$zSCORE <- (data$SCORE - meanControls)/sdControls
# grsTests <- glm(CASE ~ zSCORE + SEX + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, family="binomial", data = data)
# summary(grsTests)
# 
# 
# Call:
# glm(formula = CASE ~ zSCORE + SEX + PC1 + PC2 + PC3 + PC4 + PC5 + 
#     PC6 + PC7 + PC8 + PC9 + PC10, family = "binomial", data = data)
# 
# Deviance Residuals: 
#    Min      1Q  Median      3Q     Max  
# -1.919  -1.003  -0.810   1.278   1.796  
# 
# Coefficients:
#              Estimate Std. Error z value Pr(>|z|)    
# (Intercept)   0.43040    0.03974  10.831  < 2e-16 ***
# zSCORE        0.03021    0.01424   2.121   0.0339 *  
# SEX          -0.58575    0.02603 -22.505  < 2e-16 ***
# PC1         -35.38600    2.73911 -12.919  < 2e-16 ***
# PC2          50.17593    2.86877  17.490  < 2e-16 ***
# PC3          10.63757    2.68575   3.961 7.47e-05 ***
# PC4           0.63048    2.65991   0.237   0.8126    
# PC5          16.19265    2.70187   5.993 2.06e-09 ***
# PC6         -24.20283    2.74525  -8.816  < 2e-16 ***
# PC7           1.61607    2.62627   0.615   0.5383    
# PC8          12.66905    2.70987   4.675 2.94e-06 ***
# PC9          -5.96044    2.64009  -2.258   0.0240 *  
# PC10        -16.18338    2.65955  -6.085 1.16e-09 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# (Dispersion parameter for binomial family taken to be 1)
# 
#     Null deviance: 35275  on 26385  degrees of freedom
# Residual deviance: 34124  on 26373  degrees of freedom
# AIC: 34150
# 
# Number of Fisher Scoring iterations: 4