{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def matchAndFill (table1, table2, out):\n",
    "    df1 = pd.read_csv(table1)\n",
    "    df2 = pd.read_csv(table2)\n",
    "    \n",
    "    #df2.fillna(\"0\") doesn't actually change the df\n",
    "    for index, row in df1.iterrows():\n",
    "        snp = row[\"name\"]\n",
    "        chrom = row[\"#chrom\"].split(\"r\")\n",
    "        #print(chrom)\n",
    "        try: \n",
    "            chromNum = int(chrom[1])\n",
    "        except ValueError:\n",
    "            pass\n",
    "        for idx, rw in df2.iterrows():\n",
    "            snpToMatch = rw[\"rsID\"]\n",
    "            if snp == snpToMatch:\n",
    "                #print(rw[\"start(hg19)\"])\n",
    "                df2.at[idx, \"Chr\"] = chromNum\n",
    "                df2.at[idx, \"start(hg19)\"] = row[\"chromStart\"]\n",
    "                df2.at[idx, \"stop(hg19)\"] = row[\"chromEnd\"]\n",
    "                \n",
    "    df2.to_csv(out, index = False, header = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "table1 = \"/Volumes/T7Touch/NIHSummer2021/Data/MicrobiomeVariants/hg19_positions.csv\"\n",
    "table2 = \"/Volumes/T7Touch/NIHSummer2021/Data/MicrobiomeVariants/allMB_variants.csv\"\n",
    "out = \"/Volumes/T7Touch/NIHSummer2021/Data/MicrobiomeVariants/reformattedMB_variants.csv\"\n",
    "\n",
    "matchAndFill(table1, table2, out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdDict = {\n",
    "    'chr1' : {},\n",
    "    'chr2': {},\n",
    "    'chr3' : {},\n",
    "    'chr4': {},\n",
    "    'chr5' : {},\n",
    "    'chr6': {},\n",
    "    'chr7' : {},\n",
    "    'chr8': {},\n",
    "    'chr9' : {},\n",
    "    'chr10': {},\n",
    "    'chr11' : {},\n",
    "    'chr12': {},\n",
    "    'chr13' : {},\n",
    "    'chr14': {},\n",
    "    'chr15' : {},\n",
    "    'chr16': {},\n",
    "    'chr17' : {},\n",
    "    'chr18': {},\n",
    "    'chr19' : {},\n",
    "    'chr20': {},\n",
    "    'chr21' : {},\n",
    "    'chr22': {}\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>freq</th>\n",
       "      <th>b</th>\n",
       "      <th>se</th>\n",
       "      <th>p</th>\n",
       "      <th>N_cases</th>\n",
       "      <th>N_controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11:88249377</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9931</td>\n",
       "      <td>0.1575</td>\n",
       "      <td>0.1783</td>\n",
       "      <td>0.3771</td>\n",
       "      <td>7161</td>\n",
       "      <td>5356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:60320992</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9336</td>\n",
       "      <td>0.0605</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>0.1846</td>\n",
       "      <td>26421</td>\n",
       "      <td>442271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2:18069070</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9988</td>\n",
       "      <td>-0.6774</td>\n",
       "      <td>1.3519</td>\n",
       "      <td>0.6163</td>\n",
       "      <td>582</td>\n",
       "      <td>905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8:135908647</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2081</td>\n",
       "      <td>-0.0358</td>\n",
       "      <td>0.0273</td>\n",
       "      <td>0.1887</td>\n",
       "      <td>26421</td>\n",
       "      <td>442271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12:3871714</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9972</td>\n",
       "      <td>0.1489</td>\n",
       "      <td>1.0636</td>\n",
       "      <td>0.8886</td>\n",
       "      <td>749</td>\n",
       "      <td>658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP A1 A2    freq       b      se       p  N_cases  N_controls\n",
       "0  chr11:88249377  T  C  0.9931  0.1575  0.1783  0.3771     7161        5356\n",
       "1   chr1:60320992  A  G  0.9336  0.0605  0.0456  0.1846    26421      442271\n",
       "2   chr2:18069070  T  C  0.9988 -0.6774  1.3519  0.6163      582         905\n",
       "3  chr8:135908647  A  G  0.2081 -0.0358  0.0273  0.1887    26421      442271\n",
       "4   chr12:3871714  A  C  0.9972  0.1489  1.0636  0.8886      749         658"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3 = pd.read_csv(\"/Volumes/T7Touch/NIHSummer2021/Data/EuropeanParkinsons/nalls_dataset.csv\")\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17510617"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def testLength (dictionary):\n",
    "    size = 0\n",
    "    for key in pdDict:\n",
    "        size += len(pdDict[key])\n",
    "    print(size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "def doubleDict(df, dictName):\n",
    "    for index, row in df.iterrows():\n",
    "        chrID = row[\"SNP\"]\n",
    "        chrom = row[\"SNP\"].split(\":\")\n",
    "        #print(\"here\")\n",
    "        name = chrom[0]\n",
    "        #print(name)\n",
    "        dictName[name][chrID] = [row[\"A1\"], row[\"A2\"], row[\"freq\"], row[\"b\"], row[\"se\"], row[\"p\"], row[\"N_cases\"], row[\"N_controls\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "doubleDict(df3, pdDict)\n",
    "schedule.every(10).seconds.do(testLength(pdDict))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>chrID</th>\n",
       "      <th>Chr</th>\n",
       "      <th>start(hg19)</th>\n",
       "      <th>stop(hg19)</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>freq</th>\n",
       "      <th>b</th>\n",
       "      <th>se</th>\n",
       "      <th>p</th>\n",
       "      <th>N_cases</th>\n",
       "      <th>N_controls</th>\n",
       "      <th>Close genes</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Underlying_studies_PMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs10248138</td>\n",
       "      <td>chr7:37961001</td>\n",
       "      <td>7</td>\n",
       "      <td>37961000.0</td>\n",
       "      <td>37961001</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>EPDR1, SFRP4</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1050115</td>\n",
       "      <td>chr2:136511817</td>\n",
       "      <td>2</td>\n",
       "      <td>136511816.0</td>\n",
       "      <td>136511817</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UBXN4</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1110168</td>\n",
       "      <td>chr3:129297223</td>\n",
       "      <td>3</td>\n",
       "      <td>129297222.0</td>\n",
       "      <td>129297223</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PLXND1</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1966834</td>\n",
       "      <td>chr11:57982620</td>\n",
       "      <td>11</td>\n",
       "      <td>57982619.0</td>\n",
       "      <td>57982620</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>OR1S1</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2274669</td>\n",
       "      <td>chr20:2816821</td>\n",
       "      <td>20</td>\n",
       "      <td>2816820.0</td>\n",
       "      <td>2816821</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PCED1A</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         rsID           chrID  Chr  start(hg19)  stop(hg19) A1 A2 freq  b se  \\\n",
       "0  rs10248138   chr7:37961001    7   37961000.0    37961001  C  A    .  .  .   \n",
       "1   rs1050115  chr2:136511817    2  136511816.0   136511817  A  G    .  .  .   \n",
       "2   rs1110168  chr3:129297223    3  129297222.0   129297223  C  G    .  .  .   \n",
       "3   rs1966834  chr11:57982620   11   57982619.0    57982620  A  G    .  .  .   \n",
       "4   rs2274669   chr20:2816821   20    2816820.0     2816821  C  T    .  .  .   \n",
       "\n",
       "   p N_cases N_controls   Close genes      PMID Underlying_studies_PMID  \n",
       "0  .       .          .  EPDR1, SFRP4  33130294                26374288  \n",
       "1  .       .          .         UBXN4  33130294                26374288  \n",
       "2  .       .          .        PLXND1  33130294                26374288  \n",
       "3  .       .          .         OR1S1  33130294                26374288  \n",
       "4  .       .          .        PCED1A  33130294                26374288  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df4 = pd.read_csv(\"/Volumes/T7Touch/NIHSummer2021/Data/MicrobiomeVariants/formattedMB_variants.csv\")\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "def insertParameters(dfToFill, dictToRead):\n",
    "    for idx, row in dfToFill.iterrows():\n",
    "        chrom = \"chr\" + str(row[\"Chr\"])\n",
    "        #print(chrom)\n",
    "        chrID = row[\"chrID\"]\n",
    "        #print(pdDict[chrom][chrID][0])\n",
    "        try:\n",
    "            dfToFill.at[idx, \"A1\"] = pdDict[chrom][chrID][0]\n",
    "            dfToFill.at[idx, \"A2\"] = pdDict[chrom][chrID][1]\n",
    "            dfToFill.at[idx, \"freq\"] = pdDict[chrom][chrID][2]\n",
    "            dfToFill.at[idx, \"b\"] = pdDict[chrom][chrID][3]\n",
    "            dfToFill.at[idx, \"se\"] = pdDict[chrom][chrID][4]\n",
    "            dfToFill.at[idx, \"p\"] = pdDict[chrom][chrID][5]\n",
    "            dfToFill.at[idx, \"N_cases\"] = pdDict[chrom][chrID][6]\n",
    "            dfToFill.at[idx, \"N_controls\"] = pdDict[chrom][chrID][7]\n",
    "        except KeyError:\n",
    "            pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>chrID</th>\n",
       "      <th>Chr</th>\n",
       "      <th>start(hg19)</th>\n",
       "      <th>stop(hg19)</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>freq</th>\n",
       "      <th>b</th>\n",
       "      <th>se</th>\n",
       "      <th>...</th>\n",
       "      <th>N_cases</th>\n",
       "      <th>N_controls</th>\n",
       "      <th>Close genes</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Underlying_studies_PMID</th>\n",
       "      <th>C</th>\n",
       "      <th>A</th>\n",
       "      <th>.</th>\n",
       "      <th>G</th>\n",
       "      <th>T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs10248138</td>\n",
       "      <td>chr7:37961001</td>\n",
       "      <td>7</td>\n",
       "      <td>37961000.0</td>\n",
       "      <td>37961001</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1698</td>\n",
       "      <td>0.0402</td>\n",
       "      <td>0.0234</td>\n",
       "      <td>...</td>\n",
       "      <td>33674</td>\n",
       "      <td>449056</td>\n",
       "      <td>EPDR1, SFRP4</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>449056.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1050115</td>\n",
       "      <td>chr2:136511817</td>\n",
       "      <td>2</td>\n",
       "      <td>136511816.0</td>\n",
       "      <td>136511817</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.7983</td>\n",
       "      <td>0.0199</td>\n",
       "      <td>0.0236</td>\n",
       "      <td>...</td>\n",
       "      <td>15056</td>\n",
       "      <td>12637</td>\n",
       "      <td>UBXN4</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>12637.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1110168</td>\n",
       "      <td>chr3:129297223</td>\n",
       "      <td>3</td>\n",
       "      <td>129297222.0</td>\n",
       "      <td>129297223</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.7938</td>\n",
       "      <td>0.0249</td>\n",
       "      <td>0.0219</td>\n",
       "      <td>...</td>\n",
       "      <td>33674</td>\n",
       "      <td>449056</td>\n",
       "      <td>PLXND1</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>449056.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1966834</td>\n",
       "      <td>chr11:57982620</td>\n",
       "      <td>11</td>\n",
       "      <td>57982619.0</td>\n",
       "      <td>57982620</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.5693</td>\n",
       "      <td>-0.0275</td>\n",
       "      <td>0.0179</td>\n",
       "      <td>...</td>\n",
       "      <td>33674</td>\n",
       "      <td>449056</td>\n",
       "      <td>OR1S1</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>449056.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2274669</td>\n",
       "      <td>chr20:2816821</td>\n",
       "      <td>20</td>\n",
       "      <td>2816820.0</td>\n",
       "      <td>2816821</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2182</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>0.0237</td>\n",
       "      <td>...</td>\n",
       "      <td>33311</td>\n",
       "      <td>448891</td>\n",
       "      <td>PCED1A</td>\n",
       "      <td>33130294</td>\n",
       "      <td>26374288</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>448891.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         rsID           chrID  Chr  start(hg19)  stop(hg19) A1 A2    freq  \\\n",
       "0  rs10248138   chr7:37961001    7   37961000.0    37961001  A  C  0.1698   \n",
       "1   rs1050115  chr2:136511817    2  136511816.0   136511817  A  G  0.7983   \n",
       "2   rs1110168  chr3:129297223    3  129297222.0   129297223  C  G  0.7938   \n",
       "3   rs1966834  chr11:57982620   11   57982619.0    57982620  A  G  0.5693   \n",
       "4   rs2274669   chr20:2816821   20    2816820.0     2816821  T  C  0.2182   \n",
       "\n",
       "        b      se  ... N_cases N_controls   Close genes      PMID  \\\n",
       "0  0.0402  0.0234  ...   33674     449056  EPDR1, SFRP4  33130294   \n",
       "1  0.0199  0.0236  ...   15056      12637         UBXN4  33130294   \n",
       "2  0.0249  0.0219  ...   33674     449056        PLXND1  33130294   \n",
       "3 -0.0275  0.0179  ...   33674     449056         OR1S1  33130294   \n",
       "4  0.0123  0.0237  ...   33311     448891        PCED1A  33130294   \n",
       "\n",
       "   Underlying_studies_PMID    C    A         .    G    T  \n",
       "0                 26374288    A    C  449056.0  NaN  NaN  \n",
       "1                 26374288  NaN    A   12637.0    G  NaN  \n",
       "2                 26374288    C  NaN  449056.0    G  NaN  \n",
       "3                 26374288  NaN    A  449056.0    G  NaN  \n",
       "4                 26374288    T  NaN  448891.0  NaN    C  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "insertParameters(df4, pdDict)\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "df4.to_csv(\"/Volumes/T7Touch/NIHSummer2021/Data/MicrobiomeVariants/finalMB_variants.csv\", index = False, header = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A', 'C', 0.1698, 0.0402, 0.0234, 0.08656, 33674, 449056]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdDict[\"chr7\"][\"chr7:37961001\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'chr22:1313598'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-78-3007a93f6d9c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpdDict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"chr22\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"chr22:1313598\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m: 'chr22:1313598'"
     ]
    }
   ],
   "source": [
    "pdDict[\"chr22\"][\"chr22:1313598\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
